go/channelling/room_manager.go

```diff
@@ -209,9 +209,27 @@ func (rooms *roomManager) Get(roomID string) (room RoomWorker, ok bool) {
 	return
 }
 
+func (rooms *roomManager) isPublicRoom(roomName string) bool {
+	if rooms.PublicRoomNames != nil &&
+		rooms.PublicRoomNames.MatchString(roomName) {
+		return true
+	}
+	return false
+}
+
 func (rooms *roomManager) GetOrCreate(roomID, roomName, roomType string, credentials *DataRoomCredentials, sessionAuthenticated bool) (RoomWorker, error) {
+	isPublic := -1 // Will be evaluated lazily
```

Comment:
Please re-read all my requested changes (in code form), it will make a lot of things easier to understand and has the exact same behavior.